commit 052c40c83b4ca37be226112049b60097cb9961e1
Author: Andrew Vasquez <andrew.vasquez@qlogic.com>
Date:   Fri Jan 20 14:53:19 2006 -0800

    [SCSI] qla2xxx: Correct issue where the rport's upcall was not being made after relogin.
    
    A target can LOGO an initiator at any time (i.e. during I/O,
    due to a controller hicup, or as a simple authentication
    mechanism after an initial CDB command), when this occurs,
    the driver attempts to relogin (PLOGI) to the device via the
    DPC thread.  Add code to make the appropriate upcall to the
    FC transport layer (fc_remote_port_add()) upon successful
    completion of the PLOGI.
    
    Signed-off-by: Andrew Vasquez <andrew.vasquez@qlogic.com>
    Signed-off-by: James Bottomley <James.Bottomley@SteelEye.com>

diff --git a/drivers/scsi/qla2xxx/qla_os.c b/drivers/scsi/qla2xxx/qla_os.c
index 089e0f500363..5866a7c706a8 100644
--- a/drivers/scsi/qla2xxx/qla_os.c
+++ b/drivers/scsi/qla2xxx/qla_os.c
@@ -2250,13 +2250,8 @@ qla2x00_do_dpc(void *data)
 						DEBUG(printk("scsi(%ld): port login OK: logged in ID 0x%x\n",
 						    ha->host_no, fcport->loop_id));
 
-						fcport->port_login_retry_count =
-						    ha->port_down_retry_count * PORT_RETRY_TIME;
-						atomic_set(&fcport->state, FCS_ONLINE);
-						atomic_set(&fcport->port_down_timer,
-						    ha->port_down_retry_count * PORT_RETRY_TIME);
-
-						fcport->login_retry = 0;
+						qla2x00_update_fcport(ha,
+						    fcport);
 					} else if (status == 1) {
 						set_bit(RELOGIN_NEEDED, &ha->dpc_flags);
 						/* retry the login again */